src/main/java/org/cloudbus/cloudsim/brokers/DatacenterBrokerAbstract.java

```diff
@@ -1307,6 +1319,24 @@ public void setFailedVmsRetryDelay(final double failedVmsRetryDelay) {
         this.failedVmsRetryDelay = failedVmsRetryDelay;
     }
 
+    @Override
+    public void incrementCurrentVmCreationRetries() {
+        this.currentVmCreationRetries++;
+        if (this.currentVmCreationRetries >= this.maxVmCreationRetries) {
+            setFailedVmsRetryDelay(-1);
```

Comment:
It's not a good practice to change attributes defined by the user.
If the user sets this failedVmsRetryDelay with some value and suddenly it's changed, that causes confusion
and leads to bugs.

Instead of this if, you could change the method `isRetryFailedVms()` to:

```java
public boolean isRetryFailedVms() {
        return failedVmsRetryDelay > 0 && currentVmCreationRetries < maxVmCreationRetries;
}
```

Reply:
So here, the validation part should be moved to `isRetryFailedVms()`, and `incrementCurrentVmCreationRetries()` will only contain `this.currentVmCreationRetries++;`. Did I get this correctly? Please let me know if I am wrong. (Once I move this to new class)

Reply:
That's it.

Reply:
Also move the failedVmsRetryDelay attribute and the isRetryFailedVms method, please.

Reply:
Sure.